# Hypothesis testing

### Introduction

In this notebook, we are working in colaboration with the environmental think tank called Repair Our Air (ROA). ROA is formulating policy recommendations to improve the air quality in America, using the Environmental Protection Agency's Air Quality Index (AQI) to guide their decision making. An AQI value close to 0 signals "little to no" public health concern, while higher values are associated with increased risk to public health.

They've tasked us with leveraging AQI data to help them prioritize their strategy for improving air quality in America.

ROA is considering the following decisions. For each, we will construct a hypothesis test and an accompanying visualization, using the results of that test to make a recommendation:

1. ROA is considering a metropolitan-focused approach. Within California, they want to know if the mean AQI in Los Angeles County is statistically different from the rest of California.
2. With limited resources, ROA has to choose between New York and Ohio for their next regional office. Does New York have a lower AQI than Ohio?
3. A new policy will affect those states with a mean AQI of 10 or greater. Can we rule out Michigan from being affected by this new policy?

**Notes:**
1. For the analysis, we'll default to a 5% level of significance.
2. Throughout the lab, for two-sample t-tests, we'll use Welch's t-test (i.e., setting the `equal_var` parameter to `False` in `scipy.stats.ttest_ind()`). This will account for the possibly unequal variances between the two groups in the comparison.

In [1]:
# Import relevant libraries, packages, and modules.
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import statsmodels.api as sm
from scipy import stats

In [139]:
#import dataset
aqi= pd.read_csv(r'C:\Users\user\Desktop\Course 4\c4_epa_air_quality.csv', index_col= 0) 

## Statistical Tests
1. Formulate the null hypothesis and the alternative hypothesis.<br>
2. Set the significance level.<br>
3. Determine the appropriate test procedure.<br>
4. Compute the p-value.<br>
5. Draw your conclusion.

### Hypothesis 1:
`ROA is considering a metropolitan-focused approach. Within California, they want to know if the mean AQI in Los Angeles country is statistically different from the rest of California.`

In [149]:
#create dataframes for each sample being compared in the test
ca_la= aqi[aqi['county_name']=='Los Angeles']
ca_others =aqi[(aqi['state_name']=='California')& (aqi['county_name']!='Los Angeles')]

In [150]:
ca_la.shape

(14, 9)

In [151]:
ca_others.shape

(52, 9)

In [155]:
ca_la

date_local  state_name  county_name         city_name  \
33   2018-01-01  California  Los Angeles         Lancaster   
42   2018-01-01  California  Los Angeles     Santa Clarita   
61   2018-01-01  California  Los Angeles          Pasadena   
76   2018-01-01  California  Los Angeles       Los Angeles   
109  2018-01-01  California  Los Angeles       Los Angeles   
110  2018-01-01  California  Los Angeles       Los Angeles   
119  2018-01-01  California  Los Angeles            Reseda   
132  2018-01-01  California  Los Angeles           Compton   
163  2018-01-01  California  Los Angeles             Azusa   
172  2018-01-01  California  Los Angeles       Pico Rivera   
177  2018-01-01  California  Los Angeles        Long Beach   
189  2018-01-01  California  Los Angeles            Pomona   
233  2018-01-01  California  Los Angeles          Glendora   
250  2018-01-01  California  Los Angeles  West Los Angeles   

                   local_site_name   parameter_name   units_of_measure  \
33       Lancaster-Division Street  Carbon monoxide  Parts per million   
42                   Santa Clarita  Carbon monoxide  Parts per million   
61                        Pasadena  Carbon monoxide  Parts per million   
76                    LAX Hastings  Carbon monoxide  Parts per million   
109  Los Angeles-North Main Street  Carbon monoxide  Parts per million   
110  Los Angeles-North Main Street  Carbon monoxide  Parts per million   
119                         Reseda  Carbon monoxide  Parts per million   
132                        Compton  Carbon monoxide  Parts per million   
163                          Azusa  Carbon monoxide  Parts per million   
172                 Pico Rivera #2  Carbon monoxide  Parts per million   
177            Long Beach (Hudson)  Carbon monoxide  Parts per million   
189                         Pomona  Carbon monoxide  Parts per million   
233                       Glendora  Carbon monoxide  Parts per million   
250               West Los Angeles  Carbon monoxide  Parts per million   

     arithmetic_mean  aqi  
33          0.394737    7  
42          0.394737    7  
61          0.789474   16  
76          0.863158   17  
109         0.994737   17  
110         0.900000   16  
119         1.015789   19  
132         1.742105   40  
163         0.673684   10  
172         1.047368   18  
177         1.200000   24  
189         1.000000   20  
233         0.389474    6  
250         0.655556   11

In [154]:
ca_others

date_local  state_name     county_name                  city_name  \
16   2018-01-01  California  San Bernardino                    Ontario   
18   2018-01-01  California      Sacramento               Arden-Arcade   
26   2018-01-01  California          Orange                   La Habra   
27   2018-01-01  California         Alameda              Not in a city   
34   2018-01-01  California          Fresno                     Fresno   
40   2018-01-01  California       San Mateo               Redwood City   
43   2018-01-01  California    Contra Costa                    Concord   
45   2018-01-01  California           Butte                      Chico   
46   2018-01-01  California       Riverside                  Mira Loma   
58   2018-01-01  California            Kern                      Arvin   
62   2018-01-01  California         Alameda                    Oakland   
63   2018-01-01  California      Sacramento              Not in a city   
75   2018-01-01  California     San Joaquin                   Stockton   
77   2018-01-01  California       Riverside              Lake Elsinore   
81   2018-01-01  California   Santa Barbara  Vandenberg Air Force Base   
86   2018-01-01  California     Santa Clara                   San Jose   
87   2018-01-01  California        Humboldt                     Eureka   
99   2018-01-01  California   Santa Barbara                     Goleta   
106  2018-01-01  California        Imperial                   Calexico   
117  2018-01-01  California       San Diego                   El Cajon   
127  2018-01-01  California           Marin                 San Rafael   
130  2018-01-01  California  San Bernardino                    Fontana   
134  2018-01-01  California         Alameda                    Oakland   
139  2018-01-01  California  San Bernardino                Victorville   
141  2018-01-01  California   San Francisco              San Francisco   
147  2018-01-01  California     Santa Clara                   San Jose   
148  2018-01-01  California    Contra Costa                  San Pablo   
151  2018-01-01  California          Orange                    Anaheim   
156  2018-01-01  California          Fresno                     Clovis   
158  2018-01-01  California      Stanislaus                    Modesto   
161  2018-01-01  California            Kern                Bakersfield   
170  2018-01-01  California       San Diego                  San Diego   
171  2018-01-01  California            Inyo              Not in a city   
176  2018-01-01  California    Contra Costa              Bethel Island   
190  2018-01-01  California          Madera                     Madera   
194  2018-01-01  California  San Bernardino                     Upland   
198  2018-01-01  California       Riverside               Palm Springs   
199  2018-01-01  California       Riverside                   Rubidoux   
200  2018-01-01  California       Riverside                   Rubidoux   
203  2018-01-01  California   Santa Barbara                Santa Maria   
206  2018-01-01  California  San Bernardino             San Bernardino   
211  2018-01-01  California          Fresno              Not in a city   
213  2018-01-01  California   Santa Barbara                    Capitan   
218  2018-01-01  California          Sonoma                 Sebastopol   
220  2018-01-01  California            Napa                       Napa   
222  2018-01-01  California  San Bernardino                    Barstow   
224  2018-01-01  California      Sacramento                 Sacramento   
227  2018-01-01  California   Santa Barbara                     Lompoc   
230  2018-01-01  California        Monterey                    Salinas   
236  2018-01-01  California          Solano                    Vallejo   
239  2018-01-01  California         Alameda                    Oakland   
249  2018-01-01  California          Orange                    Anaheim   

                                       local_site_name   parameter_name  \
16     

## Formulate the hypothesis

Formulate the null and alternative hypotheses:

*   $H_0$: There is no difference in the mean AQI between Los Angeles County and the rest of California.
*   $H_A$: There is a difference in the mean AQI between Los Angeles County and the rest of California.


In [158]:
#set the significance level
significance_level= 0.05
print(significance_level)

0.05


## Determine the appropriate test procedure:

Here, we are comparing the sample means between two independent samples. Therefore, we will utilize a **two-sample  𝑡-test**.

### Compute the P-value

In [159]:
#compute the P-value
stats.ttest_ind(a= ca_la['aqi'], b= ca_others['aqi'], equal_var= False)

Ttest_indResult(statistic=2.1107010796372014, pvalue=0.049839056842410995)

### Quick Insight 

With a p-value (0.049) being less than 0.05 (as the significance level is 5% or 0.05), reject the null hypothesis in favor of the alternative hypothesis. Therefore, conclude that there is na difference in the mean of AQI between los Angeles and the rest of california.

Therefore, a metropolitan strategy may make sense in this case.

### Hypothesis 2:
`With limited resources, ROA has to choose between Newyork and Ohio for their next regional office. Does New york have a lower AQI than Ohio, and is it statistically significant?`

In [160]:
# Create dataframes for each sample being compared in the test

ny = aqi[aqi['state_name']=='New York']
ohio = aqi[aqi['state_name']=='Ohio']

In [162]:
ny.shape

(10, 9)

In [163]:
ohio.shape

(12, 9)

In [164]:
ny

date_local state_name county_name      city_name  \
90   2018-01-01   New York        Erie    Cheektowaga   
113  2018-01-01   New York       Bronx       New York   
124  2018-01-01   New York      Monroe      Rochester   
167  2018-01-01   New York    New York       New York   
173  2018-01-01   New York      Queens       New York   
182  2018-01-01   New York      Queens       New York   
184  2018-01-01   New York     Steuben  Not in a city   
195  2018-01-01   New York        Erie        Buffalo   
196  2018-01-01   New York      Monroe      Rochester   
234  2018-01-01   New York      Albany         Albany   

              local_site_name   parameter_name   units_of_measure  \
90          Buffalo Near-Road  Carbon monoxide  Parts per million   
113           PFIZER LAB SITE  Carbon monoxide  Parts per million   
124               ROCHESTER 2  Carbon monoxide  Parts per million   
167                      CCNY  Carbon monoxide  Parts per million   
173  Queens College Near Road  Carbon monoxide  Parts per million   
182          QUEENS COLLEGE 2  Carbon monoxide  Parts per million   
184       PINNACLE STATE PARK  Carbon monoxide  Parts per million   
195                   BUFFALO  Carbon monoxide  Parts per million   
196       Rochester Near-Road  Carbon monoxide  Parts per million   
234               LOUDONVILLE  Carbon monoxide  Parts per million   

     arithmetic_mean  aqi  
90          0.252632    3  
113         0.289474    3  
124         0.200000    2  
167         0.200000    2  
173         0.273684    3  
182         0.200000    2  
184         0.200000    2  
195         0.300000    3  
196         0.200000    2  
234         0.221053    3

In [165]:
ohio

date_local state_name county_name             city_name  \
1    2018-01-01       Ohio     Belmont             Shadyside   
12   2018-01-01       Ohio    Hamilton            Cincinnati   
22   2018-01-01       Ohio       Stark                Canton   
51   2018-01-01       Ohio      Summit                 Akron   
59   2018-01-01       Ohio    Cuyahoga             Cleveland   
120  2018-01-01       Ohio    Cuyahoga             Cleveland   
149  2018-01-01       Ohio    Franklin              Columbus   
191  2018-01-01       Ohio    Franklin              Columbus   
215  2018-01-01       Ohio    Cuyahoga  Warrensville Heights   
231  2018-01-01       Ohio  Montgomery                Dayton   
241  2018-01-01       Ohio      Preble             New Paris   
252  2018-01-01       Ohio    Hamilton            Cincinnati   

          local_site_name   parameter_name   units_of_measure  \
1               Shadyside  Carbon monoxide  Parts per million   
12             Taft NCore  Carbon monoxide  Parts per million   
22                 Canton  Carbon monoxide  Parts per million   
51           NIHF STEM MS  Carbon monoxide  Parts per million   
59         GT Craig NCore  Carbon monoxide  Parts per million   
120              Galleria  Carbon monoxide  Parts per million   
149              Morse Rd  Carbon monoxide  Parts per million   
191   Smoky Row Near Road  Carbon monoxide  Parts per million   
215   Cleveland Near Road  Carbon monoxide  Parts per million   
231               Reibold  Carbon monoxide  Parts per million   
241          Preble NCore  Carbon monoxide  Parts per million   
252  Cincinnati Near Road  Carbon monoxide  Parts per million   

     arithmetic_mean  aqi  
1           0.263158    5  
12          0.252632    3  
22          0.394737    6  
51          0.083333    3  
59          0.250000    3  
120         0.273684    3  
149         0.184211    3  
191         0.115789    2  
215         0.321053    5  
231         0.163158    2  
241         0.176471    2  
252         0.226316    3

## Formulate the null and alternative hypothesis


*   $H_0$: The mean AQI of New York is greater than or equal to that of Ohio.
*   $H_A$: The mean AQI of New York is **below** that of Ohio.


## Note: 
 **Significance level remains `0.05`**

#### Determine the appropriate test procedure:

Here, we are comparing the sample means between two independent samples in one direction. Therefore, you will utilize a **two-sample  𝑡-test**.

In [168]:
#Compute the P-value
tstat, p_value= stats.ttest_ind(a=ny['aqi'], b=ohio['aqi'], alternative= 'less', equal_var= False)

#print result
print(tstat)
print(p_value)

-2.025951038880333
0.03044650269193468


### Quick Insight 

With a p-value (0.030) of less than 0.05 (as the significance level is 5%) and a t-statistic < 0 (-2.036 is less than 0), **reject the null hypothesis in favor of the alternative hypothesis**.

Therefore, we can conclude at the 5% significance level that New York has a lower mean AQI than Ohio.

## Hypothesis 3: A new policy will affect those states with a mean AQI of 10 or greater. Can we rule out Michigan from being affected by this new policy

In [169]:
#create dataframes for each sample being companed in the test
michigan= aqi[aqi['state_name']== 'Michigan']

In [170]:
michigan.shape

(9, 9)

In [171]:
michigan

date_local state_name county_name      city_name  \
65   2018-01-01   Michigan       Wayne        Livonia   
122  2018-01-01   Michigan       Wayne        Detroit   
123  2018-01-01   Michigan       Wayne        Detroit   
129  2018-01-01   Michigan       Wayne        Detroit   
192  2018-01-01   Michigan       Wayne     Allen Park   
207  2018-01-01   Michigan       Wayne  Not in a city   
226  2018-01-01   Michigan        Kent   Grand Rapids   
242  2018-01-01   Michigan       Wayne        Detroit   
248  2018-01-01   Michigan       Wayne        Detroit   

              local_site_name   parameter_name   units_of_measure  \
65                 LIVONIA-NR  Carbon monoxide  Parts per million   
122               West corner  Carbon monoxide  Parts per million   
123  MARK TWAIN MIDDLE SCHOOL  Carbon monoxide  Parts per million   
129                  ELIZA-NR  Carbon monoxide  Parts per million   
192                Allen Park  Carbon monoxide  Parts per million   
207            Eliza Downwind  Carbon monoxide  Parts per million   
226                 GR-MONROE  Carbon monoxide  Parts per million   
242        (Northeast corner)  Carbon monoxide  Parts per million   
248                 NORTHWEST  Carbon monoxide  Parts per million   

     arithmetic_mean  aqi  
65          0.338889    5  
122         0.394737    8  
123         0.515789    9  
129         0.616667   11  
192         0.811111   13  
207         0.516667   10  
226         0.200000    2  
242         0.378947    7  
248         0.415789    8

### Formulate the null and alternative hypothesis:

*   $H_0$: The mean AQI of Michigan is less than or equal to 10.
*   $H_A$: The mean AQI of Michigan is greater than 10.


## Note: 
**Significance Level remains `0.05`**

### Determine the appropriate test procedure:

Here, we are comparing one sample mean relative to a particular value in one direction. Therefore, you will utilize a **one-sample  𝑡-test**.

In [176]:
#Compute the P-value
tstat,pvalue= stats.ttest_1samp(michigan['aqi'], 10, alternative= 'greater')

#print result
print(tstat)
print(pvalue)

-1.7395913343286131
0.9399405193140109


### Quick Insight 
With a p-value (0.940) being greater than 0.05 (as the significance level is 5%) and a t-statistic < 0 (-1.74), **fail to reject the null hypothesis**.

Therefore, you cannot conclude at the 5% significance level that Michigan's mean AQI is greater than 10. This implies that Michigan would not be affected by the new policy.